watcher/src/watcher.rs

```diff
@@ -360,13 +389,24 @@ impl WatcherSyncThread {
 
             // Maybe sync, maybe wait and check again.
             if is_behind {
-                let max_blocks = std::cmp::min(
-                    ledger_num_blocks - 1,
-                    lowest_next_block_to_sync + MAX_BLOCKS_PER_SYNC_ITERATION as u64,
-                );
-                watcher
-                    .sync_blocks(lowest_next_block_to_sync, Some(max_blocks), true)
+                let sync_result = watcher
+                    .sync_blocks(
+                        lowest_next_block_to_sync,
+                        Some(ledger_num_blocks - 1),
+                        Some(MAX_BLOCKS_PER_SYNC_ITERATION),
+                        true,
+                    )
                     .expect("Could not sync blocks");
+
+                // Pause for a second if a node is having trouble
+                if let SyncResult::BlockSyncError = sync_result {
+                    log::trace!(
+                        logger,
+                        "Sleeping due to sync error, lowest watcher block synced = {}... ",
+                        lowest_next_block_to_sync
```

Comment:
Nit: This message seems confusing. I'm not sure `lowest_next_block_to_sync` is relevant by this point. We can omit the number since the next iteration will recompute and log it.

```suggestion
                        "Sleeping due to sync error "
```